server.js

```diff
@@ -30,12 +32,69 @@ io.of(/^\/.*$/).on("connection", (socket) => {
   });
 
   socket.on("update", (msg) => {
+    console.log("update", msg);
     Object.assign(state[nsp], msg);
     socket.nsp.emit("state", state[nsp]);
   });
 });
 
-app.use(express.static("public"));
+// Proxy to Daily API to create rooms for breakout rooms
+app.post("/create-rooms", async (req, res) => {
+  // Add logic to check if the user is authorized to create a room
+  const isAllowed = true;
+  if (!isAllowed) {
+    return res.status(403).json({
+      error: "forbidden",
+    });
+  }
+  const dailyAPIKey = process.env.DAILY_API_KEY || "";
+  const Authorization = `Bearer ${dailyAPIKey}`;
+
+  try {
+    const headersList = {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      Authorization,
+    };
+
+    // If you'd like to dynamically change the number of rooms
+    // created, change how numberOfRooms is set.
+    const numberOfRooms = 3;
+    const roomRequets = [...Array(numberOfRooms)].map(() => {
```

Comment:
`roomRequest`

Reply:
Took me a solid 1 minute to realize you were correcting a typo. My brain was telling me "Why does Chad want me to remove the s? There's more than one request, calling it roomRequest would be silly." 🤦